core/src/main/java/ai/timefold/solver/core/impl/solver/DefaultSolverManager.java

```diff
@@ -170,10 +175,12 @@ public void terminateEarly(ProblemId_ problemId) {
     public void close() {
         solverThreadPool.shutdownNow();
         problemIdToSolverJobMap.values().forEach(DefaultSolverJob::close);
+        resourcesToRelease.forEach(DefaultSolverJob::close);
     }
 
     void unregisterSolverJob(ProblemId_ problemId) {
-        problemIdToSolverJobMap.remove(getProblemIdOrThrow(problemId));
+        var job = problemIdToSolverJobMap.remove(getProblemIdOrThrow(problemId));
+        resourcesToRelease.add(job);
```

Comment:
This creates a bit of a memory leak for as long as the solver manager is alive.
If we're unregistering the job, why not close it immediately?
We should only remember the jobs that are never unregistered.

Reply:
Also, arguably, if the job requires closing, then the job should implement `Closable` or `AutoClosable`.

Reply:
Also, Sonar reports a minor issue.